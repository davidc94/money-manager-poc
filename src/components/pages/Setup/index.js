import React, { PureComponent } from 'react';
import { ScrollView, TouchableOpacity, Text } from 'react-native';
import StyledView from '../../atoms/StyledView';
import StyledText from '../../atoms/StyledText';
import StyledButton from '../../atoms/StyledButton';

export default class Setup extends PureComponent {
  render() {
    return (
      <ScrollView>
        <TouchableOpacity onPress={() => this.props.navigation.navigate('ChooseBank')}>
          <Text>Setup</Text>
        </TouchableOpacity>
        <StyledView>
          <StyledText size weight>How does this set up work?</StyledText>
          <StyledText weight>Step 1</StyledText>
          <StyledText>
            Choose your bank -  You select your bank from the list of
            selected bank we work with!
          </StyledText>
          <StyledText weight>Step 2</StyledText>
          <StyledText>
            Your consent - We need your permission so
            that we can connect to your selected bank
            and one of the current account.
          </StyledText>
          <StyledText weight>Step 3</StyledText>
          <StyledText>
            Choose bank account - You will choose the bank account
            that you use for your day to day spend (if you have more
            than one current account)
          </StyledText>
          <StyledButton>Continue</StyledButton>
        </StyledView>
      </ScrollView>
    );
  }
}
