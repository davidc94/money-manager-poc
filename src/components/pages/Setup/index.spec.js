import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Setup from './index';

configure({ adapter: new Adapter() });

describe('setup', () => {
  const render = shallow(<Setup />);

  it('renders as expected', () => {
    expect(render).toMatchSnapshot();
  });
});