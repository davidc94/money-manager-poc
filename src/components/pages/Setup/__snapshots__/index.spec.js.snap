// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`setup renders as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Setup />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Styled(Component)
        style={
          Object {
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <StyledText
          size="base"
          weight={true}
        >
          How does this set up work?
        </StyledText>
        <StyledText
          weight={true}
        >
          Step 1
        </StyledText>
        <StyledText>
          Choose your bank -  You select your bank from the list of selected bank we work with!
        </StyledText>
        <StyledText
          style={
            Object {
              "marginBottom": 10,
              "marginTop": 20,
            }
          }
          weight={true}
        >
          Step 2
        </StyledText>
        <StyledText
          style={
            Object {
              "paddingLeft": 25,
              "paddingRight": 25,
            }
          }
        >
          Your consent - We need your permission so that we can connect to your selected bank and one of the current account.
        </StyledText>
        <StyledText
          style={
            Object {
              "marginBottom": 10,
              "marginTop": 20,
            }
          }
          weight={true}
        >
          Step 3
        </StyledText>
        <StyledText
          style={
            Object {
              "paddingLeft": 25,
              "paddingRight": 25,
            }
          }
        >
          Choose bank account - You will choose the bank account that you use for your day to day spend (if you have more than one current account)
        </StyledText>
        <StyledButton
          className="button"
          onPress={[Function]}
          style={
            Object {
              "marginTop": 25,
              "paddingLeft": 15,
              "paddingRight": 15,
            }
          }
        >
          Continue
        </StyledButton>
      </Styled(Component)>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <StyledText
            size="base"
            weight={true}
          >
            How does this set up work?
          </StyledText>,
          <StyledText
            weight={true}
          >
            Step 1
          </StyledText>,
          <StyledText>
            Choose your bank -  You select your bank from the list of selected bank we work with!
          </StyledText>,
          <StyledText
            style={
              Object {
                "marginBottom": 10,
                "marginTop": 20,
              }
            }
            weight={true}
          >
            Step 2
          </StyledText>,
          <StyledText
            style={
              Object {
                "paddingLeft": 25,
                "paddingRight": 25,
              }
            }
          >
            Your consent - We need your permission so that we can connect to your selected bank and one of the current account.
          </StyledText>,
          <StyledText
            style={
              Object {
                "marginBottom": 10,
                "marginTop": 20,
              }
            }
            weight={true}
          >
            Step 3
          </StyledText>,
          <StyledText
            style={
              Object {
                "paddingLeft": 25,
                "paddingRight": 25,
              }
            }
          >
            Choose bank account - You will choose the bank account that you use for your day to day spend (if you have more than one current account)
          </StyledText>,
          <StyledButton
            className="button"
            onPress={[Function]}
            style={
              Object {
                "marginTop": 25,
                "paddingLeft": 15,
                "paddingRight": 15,
              }
            }
          >
            Continue
          </StyledButton>,
        ],
        "style": Object {
          "flex": 1,
          "justifyContent": "center",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "How does this set up work?",
            "size": "base",
            "weight": true,
          },
          "ref": null,
          "rendered": "How does this set up work?",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Step 1",
            "weight": true,
          },
          "ref": null,
          "rendered": "Step 1",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Choose your bank -  You select your bank from the list of selected bank we work with!",
          },
          "ref": null,
          "rendered": "Choose your bank -  You select your bank from the list of selected bank we work with!",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Step 2",
            "style": Object {
              "marginBottom": 10,
              "marginTop": 20,
            },
            "weight": true,
          },
          "ref": null,
          "rendered": "Step 2",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Your consent - We need your permission so that we can connect to your selected bank and one of the current account.",
            "style": Object {
              "paddingLeft": 25,
              "paddingRight": 25,
            },
          },
          "ref": null,
          "rendered": "Your consent - We need your permission so that we can connect to your selected bank and one of the current account.",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Step 3",
            "style": Object {
              "marginBottom": 10,
              "marginTop": 20,
            },
            "weight": true,
          },
          "ref": null,
          "rendered": "Step 3",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Choose bank account - You will choose the bank account that you use for your day to day spend (if you have more than one current account)",
            "style": Object {
              "paddingLeft": 25,
              "paddingRight": 25,
            },
          },
          "ref": null,
          "rendered": "Choose bank account - You will choose the bank account that you use for your day to day spend (if you have more than one current account)",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Continue",
            "className": "button",
            "onPress": [Function],
            "style": Object {
              "marginTop": 25,
              "paddingLeft": 15,
              "paddingRight": 15,
            },
          },
          "ref": null,
          "rendered": "Continue",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [MockFunction],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Styled(Component)
          style={
            Object {
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <StyledText
            size="base"
            weight={true}
          >
            How does this set up work?
          </StyledText>
          <StyledText
            weight={true}
          >
            Step 1
          </StyledText>
          <StyledText>
            Choose your bank -  You select your bank from the list of selected bank we work with!
          </StyledText>
          <StyledText
            style={
              Object {
                "marginBottom": 10,
                "marginTop": 20,
              }
            }
            weight={true}
          >
            Step 2
          </StyledText>
          <StyledText
            style={
              Object {
                "paddingLeft": 25,
                "paddingRight": 25,
              }
            }
          >
            Your consent - We need your permission so that we can connect to your selected bank and one of the current account.
          </StyledText>
          <StyledText
            style={
              Object {
                "marginBottom": 10,
                "marginTop": 20,
              }
            }
            weight={true}
          >
            Step 3
          </StyledText>
          <StyledText
            style={
              Object {
                "paddingLeft": 25,
                "paddingRight": 25,
              }
            }
          >
            Choose bank account - You will choose the bank account that you use for your day to day spend (if you have more than one current account)
          </StyledText>
          <StyledButton
            className="button"
            onPress={[Function]}
            style={
              Object {
                "marginTop": 25,
                "paddingLeft": 15,
                "paddingRight": 15,
              }
            }
          >
            Continue
          </StyledButton>
        </Styled(Component)>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <StyledText
              size="base"
              weight={true}
            >
              How does this set up work?
            </StyledText>,
            <StyledText
              weight={true}
            >
              Step 1
            </StyledText>,
            <StyledText>
              Choose your bank -  You select your bank from the list of selected bank we work with!
            </StyledText>,
            <StyledText
              style={
                Object {
                  "marginBottom": 10,
                  "marginTop": 20,
                }
              }
              weight={true}
            >
              Step 2
            </StyledText>,
            <StyledText
              style={
                Object {
                  "paddingLeft": 25,
                  "paddingRight": 25,
                }
              }
            >
              Your consent - We need your permission so that we can connect to your selected bank and one of the current account.
            </StyledText>,
            <StyledText
              style={
                Object {
                  "marginBottom": 10,
                  "marginTop": 20,
                }
              }
              weight={true}
            >
              Step 3
            </StyledText>,
            <StyledText
              style={
                Object {
                  "paddingLeft": 25,
                  "paddingRight": 25,
                }
              }
            >
              Choose bank account - You will choose the bank account that you use for your day to day spend (if you have more than one current account)
            </StyledText>,
            <StyledButton
              className="button"
              onPress={[Function]}
              style={
                Object {
                  "marginTop": 25,
                  "paddingLeft": 15,
                  "paddingRight": 15,
                }
              }
            >
              Continue
            </StyledButton>,
          ],
          "style": Object {
            "flex": 1,
            "justifyContent": "center",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "How does this set up work?",
              "size": "base",
              "weight": true,
            },
            "ref": null,
            "rendered": "How does this set up work?",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Step 1",
              "weight": true,
            },
            "ref": null,
            "rendered": "Step 1",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Choose your bank -  You select your bank from the list of selected bank we work with!",
            },
            "ref": null,
            "rendered": "Choose your bank -  You select your bank from the list of selected bank we work with!",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Step 2",
              "style": Object {
                "marginBottom": 10,
                "marginTop": 20,
              },
              "weight": true,
            },
            "ref": null,
            "rendered": "Step 2",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Your consent - We need your permission so that we can connect to your selected bank and one of the current account.",
              "style": Object {
                "paddingLeft": 25,
                "paddingRight": 25,
              },
            },
            "ref": null,
            "rendered": "Your consent - We need your permission so that we can connect to your selected bank and one of the current account.",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Step 3",
              "style": Object {
                "marginBottom": 10,
                "marginTop": 20,
              },
              "weight": true,
            },
            "ref": null,
            "rendered": "Step 3",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Choose bank account - You will choose the bank account that you use for your day to day spend (if you have more than one current account)",
              "style": Object {
                "paddingLeft": 25,
                "paddingRight": 25,
              },
            },
            "ref": null,
            "rendered": "Choose bank account - You will choose the bank account that you use for your day to day spend (if you have more than one current account)",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Continue",
              "className": "button",
              "onPress": [Function],
              "style": Object {
                "marginTop": 25,
                "paddingLeft": 15,
                "paddingRight": 15,
              },
            },
            "ref": null,
            "rendered": "Continue",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [MockFunction],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
